### ISSUE TITLE
Multiple tagged union with undefined can't catch error 

### ISSUE THREAD

**Issue by Kwongiseok** [microsoft/TypeScript#49176]
# Bug Report

TypeScript can't catch error about Multiple tagged union (more 3 tagged union) with undefined 

### 🔎 Search Terms

Multiple tagged union , tagged union with undefined , undefined union

### 🕗 Version & Regression Information

**TypeScript Version : 4.6**

Please keep and fill in the line that best applies:
-->
- This is a bug
- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5IBQALpAA4oAqoAvKAIygA+oATC6AMxFECWAOxIEAZgEMYKAAqgA3kVCgxALlBUiAX35DRElAEFQSAB7CBCbDPmKVjBaABGqgK7mkIwUgSbtw-OMlQACEjUyRzSzl7WzZ7J1ABZwBbBwIfQT8AlABhULMLUCto1R4tMkpQAEU6UENWENZs3ngBTBIlVWr6WQ7GABpHVSYtIha22E6antsuAfjhoA)

### 💻 Code

```ts
type T = 1 | 2 | 3

interface P {
  a: T
}
interface A extends P {
  a: 1
  b: undefined
}
interface B extends P {
  a: 2
  b: number
}
interface C extends P {
  a: 3
}
type Q = A | B | C

const a: Q = { a: 1, b: 1 } // Good :  TypeScript catch error, b must be undefined

const c: Q = { a: 3, b: 1 } // Bad : TypeScript didn't catch any error, TS compiler passed it.
```

### 🙁 Actual behavior

**variable a** is error 
**variable c** is not error 

### 🙂 Expected behavior

I expected **variable c** is error when a is 3  **variable b** shouldn't exits

**Comment by MartinJohns** [comment]
> I expected variable c is error when a is 3 variable b shouldn't exits

See: https://github.com/microsoft/TypeScript/wiki/FAQ#what-is-structural-typing

Duplicate of #20863.
